src/go/k8s/api/redpanda/v1alpha2/user_types.go

```diff
@@ -108,10 +120,38 @@ type UserAuthorizationSpec struct {
 type ACLType string
 
 const (
-	ACLTypeAllow ACLType = "allow"
-	ACLTypeDeny  ACLType = "deny"
+	ACLTypeAllow   ACLType = "allow"
+	ACLTypeDeny    ACLType = "deny"
+	ACLTypeUnknown ACLType = "unknown"
 )
 
+var (
+	aclTypeFromKafka = map[kmsg.ACLPermissionType]ACLType{
+		kmsg.ACLPermissionTypeAllow: ACLTypeAllow,
+		kmsg.ACLPermissionTypeDeny:  ACLTypeDeny,
+	}
+	aclTypeToKafka = map[ACLType]kmsg.ACLPermissionType{
+		ACLTypeAllow: kmsg.ACLPermissionTypeAllow,
+		ACLTypeDeny:  kmsg.ACLPermissionTypeDeny,
+	}
+)
+
+func ACLTypeFromKafka(t kmsg.ACLPermissionType) ACLType {
```

Comment:
Just heads up, created these for a couple of reasons:

1. The `kmsg.Parse*` and corresponding typed `String` routines don't always align with the names we have (which are compatible with Strimzi enums). [See for example](https://github.com/twmb/franz-go/blob/b77dd13e2bfaee7f5181df27b40ee4a4f6a73b09/pkg/kmsg/generated.go#L46511-L46603), how it supports parsing `DescribeConfigs` through its string normalization but returns `DESCRIBE_CONFIGS` as the operation name.

2. We don't want to support every type since these are actual ACLs rather than filters (i.e. no `ACLPermissionTypeAny`)
3. Rather than returning an error, we just rely on the CRD validation to ensure we have valid perms and, in the worst case where we've missed some sort of validaiton return "unknown" types, which will fail to be synchronized by the underlying client due to not being an actual, valid type.